docs/tech/https.md

```diff
@@ -0,0 +1,111 @@
+
+- Any Python software running on Python < 2.7.9 (and note that the version of Python 2.7 provided
+  with Ubuntu 14.04 is 2.7.6. Also note that our Python Docker containers will always run the
+  latest versions of Python)
+- Vumi < 0.6.15
+- Twisted < 14.0.0
+
+### Transport Layer Security (TLS) compatibility
+
+The load balancer is currently only configured to support some versions of
+Transport Layer Security (TLS), namely TLS 1.0, 1.1 and 1.2.
+
+Some older browsers (Internet Explorer 6 or below) do not support our configuration. Once again,
+[Wikipedia has a nice table detailing compatibility](https://en.wikipedia.org/wiki/Transport_Layer_Security#Web_browsers).
+
+It's generally a good idea from a security standpoint to only support the latest TLS versions.
+We will never support technologies older than TLS 1.0 such as SSL 3.0. We may have to disable
+the older versions of TLS that we support as new vulnerabilities are found.
+
+## Enabling HTTPS on Mesos clusters
+
+The SRE team have made it really easy to set up HTTPS with free, automatically renewing
+certificates from [Let's Encrypt](https://letsencrypt.org/).
+
+### Rate limits
+
+Let's Encrypt is a free service but ultimately the service has to have some limits.
+Please read the [Let's Encrypt documentation](https://letsencrypt.org/docs/rate-limits/) for full details.
+The only limit that will be relevant to most people is the "Certificates per Registered Domain" limit.
+This limit is set at __20 certificates per week__.
+
+The way this limit works is important to understand: it is applied per registered domain. This means
+that it is easy for us to hit the rate limit if we issue certificates for many sites using a wildcard
+domain under one of Praekelt's commonly used domain names. For example, a domain like
+`seed-message-sender.seed.ng.p16n.org` is actually under the registered domain `p16n.org` - which means
+it shares its rate limit with many other Praekelt sites.
+
+tl;dr: If you have a domain that has been bought specifically for your project then you probably
+don't need to worry about rate limits. If you are using a domain that ends in `p16n.org` or `unicore.io`
+then please speak to SRE before trying to issue a certificate. Certificates are free but not unlimited.
+
+### Setup
+
+Add an application label to the app you want to set up using Mission Control. Right now
+only one domain per app is supported.
+
+The label is called `MARATHON_ACME_0_DOMAIN`. Set it to the domain name you want a
+certificate for.
+
+A few moments after adding the label the application should be available over HTTPS.
+
+### Redirecting HTTP to HTTPS
+
+Once you've verified that the application works correctly over HTTPS, set the label
+`HAPROXY_0_REDIRECT_TO_HTTPS` to `true` to redirect all HTTP traffic to HTTPS.
+
+You must verify that a certificate has been issued and HTTPS is working correctly
+before adding the redirect label.
+
+### Using HTTP Strict Transport Security
```

Comment:
The warning in my doc about HSTS is quite important. It's important people know the implications of enabling HSTS on a website.

Reply:
Added 👍